src/services/Middleware.ts

```diff
@@ -0,0 +1,56 @@
+import * as Hapi from 'hapi'
+import * as Boom from 'boom'
+import { create as createLogger } from '../common/log'
+import Config from '../services/Config'
+import SelfTest from '../services/SelfTest'
+import { Injector } from 'reduct'
+const log = createLogger('Middleware')
+
+export default class Middleware {
+  public checkSelfTestPlugin = {
+    name: 'Self Test Checker',
+    version: '1.0.0',
+    register: async (server: Hapi.Server, options: any) => {
+      server.ext('onPreAuth', (request: Hapi.Request, h: Hapi.ResponseToolkit) => {
+        const config = this.config
+        const selfTest = this.selfTest
+
+        if (!config) {
+          log.error('config missing from plugin context')
+          throw Boom.badImplementation('config missing from plugin context')
+        }
+
+        if (!selfTest) {
+          log.error('self test stats missing from plugin context')
+          throw Boom.badImplementation('self test stats missing from plugin context')
+        }
+
+        const token = request.query['token']
```

Comment:
The token should be carried in the `Authorization` header. Otherwise you might get leaks where it gets logged where it shouldn't, because nobody expects secret information to be carried via the Url